edgexfoundry/edgex-go
# feat(command): 3rd party command query via messaging

allow core command to subscribe query request and publish
commands response to 3rd party requester with external MQTT

fix #4078

Signed-off-by: Chris Hung <chris@iotechsys.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions

1. Run command service from this branch with `MESSAGEQUEUE_REQUIRED=true` environment variable
2. send query request message to `edgex/commandquery/request/<devicename>` topic.
For example send following message to `edgex/commandquery/request/Random-Boolean-Device`:
```json
{
  "correlationID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
  "apiVersion": "v2",
  "requestId": "e6e8a2f4-eb14-4649-9e2b-175247911369",
}
```
4. check the received message:
```json
{
  "ReceivedTopic": "edgex/commandquery/request/Random-Boolean-Device",
  "CorrelationID": "14a42ea6-c394-41c3-8bcd-a29b9f5e6835",
  "ApiVersion": "v2",
  "RequestID": "e6e8a2f4-eb14-4649-9e2b-175247911369",
  "ErrorCode": 0,
  "Payload": "eyJkZXZpY2VOYW1lIjoiUmFuZG9tLUJvb2xlYW4tRGV2aWNlIiwicHJvZmlsZU5hbWUiOiJSYW5kb20tQm9vbGVhbi1EZXZpY2UiLCJjb3JlQ29tbWFuZHMiOlt7Im5hbWUiOiJXcml0ZUJvb2xWYWx1ZSIsInNldCI6dHJ1ZSwicGF0aCI6Ii9hcGkvdjIvZGV2aWNlL25hbWUvUmFuZG9tLUJvb2xlYW4tRGV2aWNlL1dyaXRlQm9vbFZhbHVlIiwidXJsIjoiaHR0cDovL2VkZ2V4LWNvcmUtY29tbWFuZDo1OTg4MiIsInBhcmFtZXRlcnMiOlt7InJlc291cmNlTmFtZSI6IkJvb2wiLCJ2YWx1ZVR5cGUiOiJCb29sIn0seyJyZXNvdXJjZU5hbWUiOiJFbmFibGVSYW5kb21pemF0aW9uX0Jvb2wiLCJ2YWx1ZVR5cGUiOiJCb29sIn1dfSx7Im5hbWUiOiJXcml0ZUJvb2xBcnJheVZhbHVlIiwic2V0Ijp0cnVlLCJwYXRoIjoiL2FwaS92Mi9kZXZpY2UvbmFtZS9SYW5kb20tQm9vbGVhbi1EZXZpY2UvV3JpdGVCb29sQXJyYXlWYWx1ZSIsInVybCI6Imh0dHA6Ly9lZGdleC1jb3JlLWNvbW1hbmQ6NTk4ODIiLCJwYXJhbWV0ZXJzIjpbeyJyZXNvdXJjZU5hbWUiOiJCb29sQXJyYXkiLCJ2YWx1ZVR5cGUiOiJCb29sQXJyYXkifSx7InJlc291cmNlTmFtZSI6IkVuYWJsZVJhbmRvbWl6YXRpb25fQm9vbEFycmF5IiwidmFsdWVUeXBlIjoiQm9vbCJ9XX0seyJuYW1lIjoiQm9vbCIsImdldCI6dHJ1ZSwic2V0Ijp0cnVlLCJwYXRoIjoiL2FwaS92Mi9kZXZpY2UvbmFtZS9SYW5kb20tQm9vbGVhbi1EZXZpY2UvQm9vbCIsInVybCI6Imh0dHA6Ly9lZGdleC1jb3JlLWNvbW1hbmQ6NTk4ODIiLCJwYXJhbWV0ZXJzIjpbeyJyZXNvdXJjZU5hbWUiOiJCb29sIiwidmFsdWVUeXBlIjoiQm9vbCJ9XX0seyJuYW1lIjoiQm9vbEFycmF5IiwiZ2V0Ijp0cnVlLCJzZXQiOnRydWUsInBhdGgiOiIvYXBpL3YyL2RldmljZS9uYW1lL1JhbmRvbS1Cb29sZWFuLURldmljZS9Cb29sQXJyYXkiLCJ1cmwiOiJodHRwOi8vZWRnZXgtY29yZS1jb21tYW5kOjU5ODgyIiwicGFyYW1ldGVycyI6W3sicmVzb3VyY2VOYW1lIjoiQm9vbEFycmF5IiwidmFsdWVUeXBlIjoiQm9vbEFycmF5In1dfV19",
  "ContentType": "application/json",
  "QueryParams": null
}
```

## New Dependency Instructions (If applicable)
